queries.sql

```diff
@@ -1,3 +1,103 @@
+UPDATE ANIMALS SET weight_kg = -1 * weight_kg;
+ROLLBACK TO SAVEPOINT delete_after_january_2022;
+UPDATE ANIMALS SET weight_kg = -1 * weight_kg WHERE weight_kg < 0;
+COMMIT;
+
+-- Aggregate Transactions
+
+-- How many animals are there?
+SELECT count(animals) FROM animals; -- 10
+
+
+-- How many animals have never tried to escape?
+SELECT count(animals) FROM animals WHERE escape_attempts = 0; -- 2
+
+
+-- What is the average weight of animals?
+SELECT avg(weight_kg) FROM animals; -- 15.55000
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT SUM(escape_attempts) FROM animals WHERE neutered; -- 20
+SELECT SUM(escape_attempts) FROM animals WHERE NOT neutered; -- 4
+
+
+-- What is the minimum and maximum weight of each type of animal?
+-- Minimum of type digimon
+SELECT MIN(weight_kg) FROM animals WHERE species = 'digimon'; -- 5.7
+-- Maximum of type digimon
+SELECT MAX(weight_kg) FROM animals WHERE species = 'digimon'; -- 45
+-- Minimum of type pokemon
+SELECT MIN(weight_kg) FROM animals WHERE species = 'pokemon'; -- 11
+-- Maximum of type pokemon
+SELECT MAX(weight_kg) FROM animals WHERE species = 'pokemon'; -- 17
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+-- Average escape attempts of type digimon
+SELECT AVG(escape_attempts) FROM animals WHERE species = 'digimon' AND date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'; -- 0
+-- Average escape attempts of type pokemon
+SELECT AVG(escape_attempts) FROM animals WHERE species = 'pokemon' AND date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'; -- 3.0
+
+/* Write queries (using JOIN) to answer the following questions: */
+
+
+-- What animals belong to Melody Pond?
+
+SELECT * FROM animals INNER JOIN owners ON animals.owner_id = owners.id WHERE owners.full_name = 'Melody Pond';
+
+-- List of all animals that are pokemon (their type is Pokemon)
+
+SELECT * FROM animals INNER JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
+
+-- List all owners and their animals, remember to include those that don't own any animal.
+
+SELECT animals.name, owners.full_name FROM owners INNER JOIN animals ON animals.owner_id = owners.id WHERE owners.full_name = 'Sam Smith';
+
+SELECT animals.name, owners.full_name FROM owners INNER JOIN animals ON animals.owner_id = owners.id WHERE owners.full_name = 'Jennifer Orwell';
+
+SELECT animals.name, owners.full_name FROM owners INNER JOIN animals ON animals.owner_id = owners.id WHERE owners.full_name = 'Bob';
+
+SELECT animals.name, owners.full_name FROM owners INNER JOIN animals ON animals.owner_id = owners.id WHERE owners.full_name = 'Melody Pond';
+
+SELECT animals.name, owners.full_name FROM owners INNER JOIN animals ON animals.owner_id = owners.id WHERE owners.full_name = 'Dean Winchester';
+
+SELECT animals.name, owners.full_name FROM owners INNER JOIN animals ON animals.owner_id = owners.id WHERE owners.full_name = 'Jodie Whittaker';
```

Comment:
- [ ] I can't see these queries
-  List all owners and their animals, remember to include those that don't own any animal.
-  How many animals are there per species?

Reply:
well resolved! ✔️